### ISSUE TITLE
Nextjs build fails. [Error: ENOTEMPTY: directory not empty, rmdir '/codebuild/output/src615130554/src/nextpo/apps/next/node_modules/eslint-plugin-jsx-a11y/node_modules/.bin']

### ISSUE THREAD

**Issue by MiguelNiblock** [aws-amplify/amplify-hosting#2775]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

arn:aws:amplify:us-east-1:224875105023:apps/d11vufdm6y7riq

### Region

us-east-1

### Amplify Hosting feature

Not applicable

### Describe the bug

When the nextjs app is deployed and building the front-end phase, the command `next build` fails. This command is successful when I run it locally.

This is the error log

```
2022-05-30T07:42:25.501Z [INFO]: [4/4] Building fresh packages...
2022-05-30T07:42:25.833Z [INFO]: Done in 57.34s.
2022-05-30T07:42:25.872Z [INFO]: # Completed phase: preBuild
                                 # Starting phase: build
2022-05-30T07:42:25.872Z [INFO]: # Executing command: cd apps/next && yarn build
2022-05-30T07:42:26.015Z [INFO]: yarn run v1.22.0
2022-05-30T07:42:26.043Z [INFO]: $ next build
2022-05-30T07:42:27.040Z [INFO]: info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
2022-05-30T07:42:27.469Z [INFO]: warn  - No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
2022-05-30T07:42:27.487Z [INFO]: Attention: Next.js now collects completely anonymous telemetry regarding usage.
2022-05-30T07:42:27.487Z [INFO]: This information is used to shape Next.js' roadmap and prioritize features.
                                 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
                                 https://nextjs.org/telemetry
2022-05-30T07:42:27.677Z [INFO]: info  - Checking validity of types...
2022-05-30T07:42:32.549Z [INFO]: info  - Creating an optimized production build...
2022-05-30T07:42:32.600Z [WARNING]: 
2022-05-30T07:42:32.600Z [WARNING]: > Build error occurred
2022-05-30T07:42:32.601Z [WARNING]: [Error: ENOTEMPTY: directory not empty, rmdir '/codebuild/output/src615130554/src/nextpo/apps/next/node_modules/eslint-plugin-jsx-a11y/node_modules/.bin'] {
                                    errno: -39,
                                    code: 'ENOTEMPTY',
                                    syscall: 'rmdir',
                                    path: '/codebuild/output/src615130554/src/nextpo/apps/next/node_modules/eslint-plugin-jsx-a11y/node_modules/.bin'
                                    }
2022-05-30T07:42:32.629Z [WARNING]: error Command failed with exit code 1.
2022-05-30T07:42:32.629Z [INFO]: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2022-05-30T07:42:32.637Z [ERROR]: !!! Build failed
2022-05-30T07:42:32.638Z [ERROR]: !!! Non-Zero Exit Code detected
2022-05-30T07:42:32.638Z [INFO]: # Starting environment caching...
2022-05-30T07:42:32.639Z [INFO]: # Environment caching completed
Terminating logging... 

```

### Expected behavior

The nextjs build should succeed.

### Reproduction steps

Connect the github repository to amplify hosting using amplify console. https://github.com/MiguelNiblock/nextpo/tree/main
Deploy and watch logs.

### Build Settings

```yaml
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - yarn
    build:
      commands:
        - cd apps/next && yarn build
  artifacts:
    baseDirectory: apps/next/.next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

### Additional information

Im trying to deploy a nextjs app to amplify. The same app is successfully deployed to Vercel, and the build command succeeds locally. It only fails in amplify.

This is the deployed app in Vercel. https://nextpo-zgdr.vercel.app/

**Comment by hloriana** [comment]
Hi @MiguelNiblock 👋🏽  thanks for providing details and repro steps. We will try to reproduce this and find the solution. In the mean time, try changing `baseDirectory: apps/next/.next` to `baseDirectory: .next`. Since you `cd` into `apps/next` in the build phase, that should be the current location of `.next`.

**Comment by MiguelNiblock** [comment]
@hloriana thank you for looking at this issue. I adjusted the baseDirectory as you suggested, and I got the same error.
```
2022-06-01T04:07:19.142Z [INFO]: [4/4] Building fresh packages...
2022-06-01T04:07:19.489Z [INFO]: Done in 60.36s.
2022-06-01T04:07:19.524Z [INFO]: # Completed phase: preBuild
                                 # Starting phase: build
2022-06-01T04:07:19.524Z [INFO]: # Executing command: cd apps/next && yarn build
2022-06-01T04:07:19.673Z [INFO]: yarn run v1.22.0
2022-06-01T04:07:19.700Z [INFO]: $ next build
2022-06-01T04:07:20.733Z [INFO]: info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
2022-06-01T04:07:21.172Z [INFO]: warn  - No build cache found. Please configure build caching for faster rebuilds. Read more: https://nextjs.org/docs/messages/no-cache
2022-06-01T04:07:21.188Z [INFO]: Attention: Next.js now collects completely anonymous telemetry regarding usage.
2022-06-01T04:07:21.188Z [INFO]: This information is used to shape Next.js' roadmap and prioritize features.
                                 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
                                 https://nextjs.org/telemetry
2022-06-01T04:07:21.387Z [INFO]: info  - Checking validity of types...
2022-06-01T04:07:26.329Z [INFO]: info  - Creating an optimized production build...
2022-06-01T04:07:26.385Z [WARNING]: 
2022-06-01T04:07:26.385Z [WARNING]: > Build error occurred
2022-06-01T04:07:26.387Z [WARNING]: [Error: ENOTEMPTY: directory not empty, rmdir '/codebuild/output/src776904929/src/nextpo/apps/next/node_modules/eslint-plugin-jsx-a11y/node_modules/.bin'] {
                                    errno: -39,
                                    code: 'ENOTEMPTY',
                                    syscall: 'rmdir',
                                    path: '/codebuild/output/src776904929/src/nextpo/apps/next/node_modules/eslint-plugin-jsx-a11y/node_modules/.bin'
                                    }
2022-06-01T04:07:26.416Z [WARNING]: error Command failed with exit code 1.
2022-06-01T04:07:26.417Z [INFO]: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2022-06-01T04:07:26.425Z [ERROR]: !!! Build failed
2022-06-01T04:07:26.426Z [ERROR]: !!! Non-Zero Exit Code detected
2022-06-01T04:07:26.426Z [INFO]: # Starting environment caching...
2022-06-01T04:07:26.426Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by Sajithrajan03** [comment]
```
 rm -rf node_modules
 npm i
```

And then try building it. 
`npm run build`
Worked for me.

**Comment by ravindrakanapuram** [comment]
HI , 
I have faced the same bug the issue with the node version need to use the latest version of the nodeJS 😊

initially i have tested with docker image buid it had successfull build results for nextjs app on left side shell in image ,  in the localhost to used the node v21.7.3 () to build it was failed give the error on right side shell
`npm run build`   

![Image](https://github.com/user-attachments/assets/5aab246e-36fc-450f-acb2-bd5666c9c38c)

here is the succesful build result with latest node version 23.6.0(LTS)

![Image](https://github.com/user-attachments/assets/1f3131b8-1e5f-4edd-ac8f-3252a36c87c8)


it is better to upgrade their node version in aws amplify

**Comment by alirezarastineh** [comment]
I also only needed to update my Node to the latest (23.6.0) version and it's solved the error
